DMCompiler/DM/Visitors/DMProcBuilder.cs

```diff
@@ -129,25 +129,77 @@ class DMProcBuilder {
             _proc.Break(statementBreak.Label);
         }
 
-        public void ProcessStatementSet(DMASTProcStatementSet statementSet) {
-            //TODO: Proc attributes
+        public void ProcessStatementSet(DMASTProcStatementSet statementSet)
+        {
+            var attribute = statementSet.Attribute.ToLower();
+            if (!DMExpression.TryConstant(_dmObject, _proc, statementSet.Value, out var constant)) {
+                throw new CompileErrorException(statementSet.Location, $"{attribute} attribute should be a constant");
+            }
+
             switch (statementSet.Attribute.ToLower()) {
                 case "waitfor": {
-                    if (!DMExpression.TryConstant(_dmObject, _proc, statementSet.Value, out var constant)) {
-                        throw new CompileErrorException(statementSet.Location, $"waitfor attribute should be a constant");
+                    if (constant.IsTruthy())
+                    {
+                        // "waitfor" defaults to enabled
+                        _proc.Attributes &= ~ProcAttributes.DisableWaitfor;
+                    }
+                    else
+                    {
+                        _proc.Attributes |= ProcAttributes.DisableWaitfor;
                     }
```

Comment:
There is a method on DMProc that does this
```suggestion
                    _proc.WaitFor(constant.IsTruthy());
```